modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerIndexingTest.java

```diff
@@ -85,7 +90,111 @@ public void testValidateIndexesFailedOnNotIdleCluster() throws Exception {
+    }
+
+    /** Run index validation check on busy cluster. */
+    @Test
+    public void testIdleVerifyCheckFailsOnNotIdleClusterWithOverwriteWithoutPers() throws Exception {
+        persistenceEnable(false);
+
+        runIdleVerifyCheckCrcFailsOnNotIdleCluster(true);
+    }
+
+    /** Run index validation check on busy cluster. */
+    @Test
+    public void testIdleVerifyCheckFailsOnNotIdleClusterWithoutOverwriteWithPers() throws Exception {
+        runIdleVerifyCheckCrcFailsOnNotIdleCluster(false);
+    }
+
+    /** Run index validation check on busy cluster. */
+    @Test
+    public void testIdleVerifyCheckFailsOnNotIdleClusterWithoutOverwriteWithoutPers() throws Exception {
+        persistenceEnable(false);
+
+        runIdleVerifyCheckCrcFailsOnNotIdleCluster(false);
+    }
+
+    /**
+     * Check idle on busy cluster.
+     *
+     * @param allowOverwrite Overwrite param for datastreamer.
+     * @throws Exception
+     */
+    public void runIdleVerifyCheckCrcFailsOnNotIdleCluster(boolean allowOverwrite) throws Exception {
+        IgniteEx ig = startGrids(1);
+
+        ig.cluster().active(true);
+
+        createCacheAndPreload(ig, 0);
+
+        GridCacheDatabaseSharedManager db = null;
+
+        if (persistenceEnable()) {
+            forceCheckpoint();
+
+            db = (GridCacheDatabaseSharedManager) ig.context().cache().context().database();
+            db.enableCheckpoints(false).get();
+        }
+
+        AtomicBoolean stopFlag = new AtomicBoolean();
+
+        IgniteInternalFuture<Object> f = runAsync(new Runnable() {
+            @Override public void run() {
+                try (IgniteDataStreamer<Object, Object> ldr = ig.dataStreamer(DEFAULT_CACHE_NAME)) {
+                    ldr.allowOverwrite(allowOverwrite);
+
+                    boolean addFlag = true;
+
+                    int i = 0;
+
+                    while (!stopFlag.get()) {
+                        if (addFlag)
+                            ldr.addData(i, i);
+                        else
+                            ldr.removeData(i);
+
+                        if (++i == 100000) {
+                            ldr.flush();
+
+                            addFlag = !addFlag;
+
+                            i = 0;
+                        }
+                    }
+                }
+            }
+        });
+
+        injectTestSystemOut();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "validate_indexes", "--check-crc"));
+
+        stopFlag.set(true);
+
+        f.get();
+
+        String out = testOut.toString();
+
+        assertContains(log, out, IDLE_DATA_ALTERATION_MSG);
+
+        testOut.reset();
+
+        if (persistenceEnable())
+            db.enableCheckpoints(true).get();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "validate_indexes", "--check-crc"));
+
+        out = testOut.toString();
+
+        assertNotContains(log, out, IDLE_DATA_ALTERATION_MSG);
```

Comment:
You log group names for cache groups associated with diverged partitions but never check them in log